lib/spack/spack/test/cmd/build_env.py

```diff
@@ -58,6 +63,31 @@ def test_pickle(tmpdir):
         assert "PATH" in environment
 
 
+# TODO params [i, b, c] require a spec that has proceeded with a directory
+# TODO praram [e] requires an active env
+@pytest.mark.parametrize("cd_key", ["r", "spack-root"])
+@pytest.mark.usefixtures("config", "mock_packages", "working_env")
+def test_cd(cd_key, tmpdir, monkeypatch, capfd):
+    """test that a subshell will navigate using spack cd before running commands"""
+    cmd = "pwd" if sys.platform != "win32" else 'powershell.exe -Command "& {(Get-Location).Path}"'
+
+    def mock_execvp(_, args):
+        """os.execvp will kill take over the pytest process when it is successful"""
+        result = subprocess.check_output(args, universal_newlines=True)
+        print(result)
+
+    with tmpdir.as_cwd():
+        monkeypatch.setattr(os, "execvp", mock_execvp)
+
+        pwd = os.getcwd()
+        spec = Spec("zlib").concretized()
+        run_command_in_subshell(spec, Context.BUILD, [cmd], cd_arg=cd_key)
```

Comment:
```suggestion
        run_command_in_subshell(spec, Context.BUILD, cmd, cd_arg=cd_key)
```

Reply:
`subprocess.check_output` takes either a list of command + args or a string with the whole invocation, composing the Windows command line into a list causes the entire powershell invocation to be interpreted as a single command, which of course cannot be found.